### ISSUE TITLE
Running UI tests on iOS 9.3 together with newer simulators fails

### ISSUE THREAD

**Issue by terje** [fastlane/fastlane#11929]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

There seems to be an issue with `run_tests` when running on iOS 9.3 simulators together with newer simulator versions. The iOS 9.3 simulator gets stuck in a restart loop trying to launch the app and the UI tests never finish.

I have created a simple new single view app that exhibits this odd behaviour: https://github.com/terje/fastlane-test-bug-reproduction

Running the UI tests on an iOS 10.3.1 simulator and an iOS 11 simulator in parallel works just fine, whereas running the tests on iOS 9.3 and iOS 10.3.1 in parallel fails. Also, running the tests on an iOS 9.3 simulator alone works fine.

While the app is stuck in the restart loop, the device log is showing a lot of entries like this:

<details>
<pre>
Feb 26 09:46:53 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xc68b][67182]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:53 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x4a2d][67183]): Program specified by service does not contain one of the requested architectures: 
Feb 26 09:46:53 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x3436][67184]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:54 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x67e4][67185]): Program specified by service does not contain one of the requested architectures: 
Feb 26 09:46:54 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xd614][67186]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:54 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x7c64][67188]): Program specified by service does not contain one of the requested architectures: 
Feb 26 09:46:54 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xca83][67189]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:55 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x13c2][67190]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:56 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xd7bb][67192]): Program specified by service does not contain one of the requested architectures: 
Feb 26 09:46:56 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xfe3d][67193]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:56 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x60f2][67195]): Program specified by service does not contain one of the requested architectures: 
Feb 26 09:46:56 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xd739][67196]): Program specified by service does not contain one of the requested architectures: h!
Feb 26 09:46:57 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xd1ba][67200]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0
Feb 26 09:46:57 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0xe3dc][67202]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0
Feb 26 09:46:57 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x93ae][67230]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0
Feb 26 09:46:57 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x6f3f][67231]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0
Feb 26 09:46:58 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x21a7][67232]): Program specified by service does not contain one of the requested architectures: Ð×
Feb 26 09:46:58 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x4ed2][67233]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0
Feb 26 09:46:58 Mac-mini com.apple.CoreSimulator.SimDevice.140DF40C-97ED-4401-8440-C6B1FF67E01A[66074] (UIKitApplication:no.terje.FastlaneBugApp[0x34a0][67234]): Program specified by service does not contain one of the requested architectures: XPC_FLAGS=0x0

</pre>
</details>

##### Complete output when running fastlane, including the stack trace and command used

This is from the test run exhibiting the bug. I had to CTRL+C after a while:

<details>
  <pre>
[10:22:26]: [32m------------------------------[0m
[10:22:26]: [32m--- Step: default_platform ---[0m
[10:22:26]: [32m------------------------------[0m
[10:22:26]: [32mDriving the lane 'ios test' 🚀[0m
[10:22:26]: [32m-----------------------[0m
[10:22:26]: [32m--- Step: run_tests ---[0m
[10:22:26]: [32m-----------------------[0m
[10:22:26]: [36m$ xcodebuild -list -project ./FastlaneBugApp.xcodeproj[0m
[10:22:27]: [36m$ xcodebuild -showBuildSettings -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj[0m
[10:22:29]: [36m$ xcodebuild -list -project ./FastlaneBugApp.xcodeproj[0m
[10:22:30]: [36m$ xcodebuild -showBuildSettings -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj[0m

+------------------------+----------------------------------------------------------------------------------------+
|                                             [32mSummary for scan 2.82.0[0m                                             |
+------------------------+----------------------------------------------------------------------------------------+
| devices                | ["iPhone 5 (9.3)", "iPhone 6 (10.3.1)"]                                                |
| project                | ./FastlaneBugApp.xcodeproj                                                             |
| scheme                 | FastlaneBugApp                                                                         |
| derived_data_path      | /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxf  |
|                        | ijcexhcerfzxlwdc                                                                       |
| clean                  | false                                                                                  |
| skip_build             | false                                                                                  |
| output_directory       | ./fastlane/test_output                                                                 |
| output_types           | html,junit                                                                             |
| buildlog_path          | ~/Library/Logs/scan                                                                    |
| include_simulator_logs | false                                                                                  |
| open_report            | false                                                                                  |
| skip_slack             | false                                                                                  |
| slack_only_on_failure  | false                                                                                  |
| use_clang_report_name  | false                                                                                  |
| fail_build             | true                                                                                   |
| xcode_path             | /Applications/Xcode.app                                                                |
+------------------------+----------------------------------------------------------------------------------------+

[10:22:31]: [36m$ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj -destination 'platform=iOS Simulator,id=140DF40C-97ED-4401-8440-C6B1FF67E01A' -destination 'platform=iOS Simulator,id=4BD7096A-68AA-4E58-900D-80CA87F5B823' -derivedDataPath '/Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc' build test | tee '/Users/terje.tjervaag/Library/Logs/scan/FastlaneBugApp-FastlaneBugApp.log' | xcpretty  --report html --output '/Users/terje.tjervaag/Development/FastlaneBugApp/fastlane/test_output/report.html' --report junit --output '/Users/terje.tjervaag/Development/FastlaneBugApp/fastlane/test_output/report.junit' --report junit --output '/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/junit_report20180226-70818-8dmmzq'[0m
[10:22:31]: ▸ [35mLoading...[0m
[10:22:33]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:22:33]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:22:33]: [35m[33m▸[0m [39;1mLinking[0m FastlaneBugApp[0m
[10:22:33]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:22:34]: [35mRunning Tests: [33m▸[0m [39;1mTouching[0m FastlaneBugApp.app[0m
[10:22:34]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:22:35]: [35m[33m▸[0m [39;1mBuild[0m Succeeded[0m
[10:22:35]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:22:35]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:22:35]: [35m[33m▸[0m [39;1mLinking[0m FastlaneBugApp[0m
[10:22:35]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:22:36]: [35mRunning Tests: [33m▸[0m [39;1mTouching[0m FastlaneBugApp.app[0m
[10:22:36]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:22:36]: [35m[33m▸[0m [39;1mBuild[0m Succeeded[0m
[10:22:36]: ▸ [35m2018-02-26 10:22:36.870 xcodebuild[70860:2296153]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:22:36]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-6F9A685D-BE10-4D3E-B359-98F3D5B353C5/FastlaneBugAppUITests-95D75872-39EA-41DC-9B5E-FB2C5A2A2EE9/Session-FastlaneBugAppUITests-2018-02-26_102236-5hYtKL.log[0m
[10:22:36]: ▸ [35m2018-02-26 10:22:36.871 xcodebuild[70860:2295997] [MT] IDETestOperationsObserverDebug: (B1116D6E-DF9F-4E53-B649-37A8DD3A66CC) Beginning test session FastlaneBugAppUITests-B1116D6E-DF9F-4E53-B649-37A8DD3A66CC at 2018-02-26 10:22:36.870 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fd57d4a8df0> {[0m
[10:22:36]: ▸ [35mSimDevice: iPhone 5 (140DF40C-97ED-4401-8440-C6B1FF67E01A, iOS 9.3, Booted)[0m
[10:22:36]: ▸ [35m} (9.3 (13E233))[0m
[10:22:36]: ▸ [35m2018-02-26 10:22:36.879 xcodebuild[70860:2296091]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:22:36]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-6F9A685D-BE10-4D3E-B359-98F3D5B353C5/FastlaneBugAppTests-867AC704-78D9-4E4C-A880-C0A5F31225B1/Session-FastlaneBugAppTests-2018-02-26_102236-LBiNIU.log[0m
[10:22:36]: ▸ [35m2018-02-26 10:22:36.879 xcodebuild[70860:2295997] [MT] IDETestOperationsObserverDebug: (6DADA4C8-0B28-48ED-8C64-C4961F4CB127) Beginning test session FastlaneBugAppTests-6DADA4C8-0B28-48ED-8C64-C4961F4CB127 at 2018-02-26 10:22:36.879 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fd57d4a8df0> {[0m
[10:22:36]: ▸ [35mSimDevice: iPhone 5 (140DF40C-97ED-4401-8440-C6B1FF67E01A, iOS 9.3, Booted)[0m
[10:22:36]: ▸ [35m} (9.3 (13E233))[0m
[10:22:37]: ▸ [35m2018-02-26 10:22:37.127 xcodebuild[70860:2296153]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:22:37]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-75E1E15A-C033-4EE4-8CF3-A7071D849836/FastlaneBugAppUITests-B4C593AE-A08A-42AB-9738-08EAC046431B/Session-FastlaneBugAppUITests-2018-02-26_102237-fe3Pvr.log[0m
[10:22:37]: ▸ [35m2018-02-26 10:22:37.128 xcodebuild[70860:2295997] [MT] IDETestOperationsObserverDebug: (2341A60E-81CA-4FCC-808B-439DAA3DC909) Beginning test session FastlaneBugAppUITests-2341A60E-81CA-4FCC-808B-439DAA3DC909 at 2018-02-26 10:22:37.127 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fd57d4cd6c0> {[0m
[10:22:37]: ▸ [35mSimDevice: iPhone 6 (4BD7096A-68AA-4E58-900D-80CA87F5B823, iOS 10.3, Booted)[0m
[10:22:37]: ▸ [35m} (10.3.1 (14E8301))[0m
[10:22:37]: ▸ [35m2018-02-26 10:22:37.134 xcodebuild[70860:2296177]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:22:37]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-75E1E15A-C033-4EE4-8CF3-A7071D849836/FastlaneBugAppTests-EC2330F9-514E-4DC1-92F4-92C88BEB8600/Session-FastlaneBugAppTests-2018-02-26_102237-0FPa8a.log[0m
[10:22:37]: ▸ [35m2018-02-26 10:22:37.134 xcodebuild[70860:2295997] [MT] IDETestOperationsObserverDebug: (12D6D373-BAC3-42DE-870D-A0F203721063) Beginning test session FastlaneBugAppTests-12D6D373-BAC3-42DE-870D-A0F203721063 at 2018-02-26 10:22:37.134 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fd57d4cd6c0> {[0m
[10:22:37]: ▸ [35mSimDevice: iPhone 6 (4BD7096A-68AA-4E58-900D-80CA87F5B823, iOS 10.3, Booted)[0m
[10:22:37]: ▸ [35m} (10.3.1 (14E8301))[0m
[10:22:37]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:22:37]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:22:38]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:22:38]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:22:39]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugAppTests [Debug][0m
[10:22:39]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:22:39]: [35m[33m▸[0m [39;1mLinking[0m FastlaneBugAppTests[0m
[10:22:40]: [35mRunning Tests: [33m▸[0m [39;1mTouching[0m FastlaneBugAppTests.xctest[0m
[10:22:40]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app/PlugIns/FastlaneBugAppTests.xctest[0m
[10:22:41]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugAppUITests [Debug][0m
[10:22:41]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
+----------------------------+----------------------------------------------------------------------------------------+
|                                                    [33mLane Context[0m                                                     |
+----------------------------+----------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM           | ios                                                                                    |
| PLATFORM_NAME              | ios                                                                                    |
| LANE_NAME                  | ios test                                                                               |
| SCAN_DERIVED_DATA_PATH     | /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxf  |
|                            | ijcexhcerfzxlwdc                                                                       |
| SCAN_GENERATED_PLIST_FILES | ["/Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrm  |
|                            | xfijcexhcerfzxlwdc/Logs/Test/2DA75DC8-470B-45B3-A0F4-1639F38EE8C5_TestSummaries.plist  |
|                            | "]                                                                                     |
| SCAN_GENERATED_PLIST_FILE  | /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxf  |
|                            | ijcexhcerfzxlwdc/Logs/Test/2DA75DC8-470B-45B3-A0F4-1639F38EE8C5_TestSummaries.plist    |
+----------------------------+----------------------------------------------------------------------------------------+

+------+------------------+-------------+
|           [32mfastlane summary[0m            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 2    | run_tests        | 90          |
+------+------------------+-------------+

[10:23:56]: [31mfastlane finished with errors[0m

Cancelled... use --verbose to show the stack trace

</pre>
</details>

This is from a working test run:

<details>
  <pre>
[10:26:24]: [32m------------------------------[0m
[10:26:24]: [32m--- Step: default_platform ---[0m
[10:26:24]: [32m------------------------------[0m
[10:26:24]: [32mDriving the lane 'ios workingtest' 🚀[0m
[10:26:24]: [32m-----------------------[0m
[10:26:24]: [32m--- Step: run_tests ---[0m
[10:26:24]: [32m-----------------------[0m
[10:26:24]: [36m$ xcodebuild -list -project ./FastlaneBugApp.xcodeproj[0m
[10:26:25]: [36m$ xcodebuild -showBuildSettings -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj[0m
[10:26:27]: [36m$ xcodebuild -list -project ./FastlaneBugApp.xcodeproj[0m
[10:26:28]: [36m$ xcodebuild -showBuildSettings -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj[0m

+------------------------+----------------------------------------------------------------------------------------+
|                                             [32mSummary for scan 2.82.0[0m                                             |
+------------------------+----------------------------------------------------------------------------------------+
| devices                | ["iPhone 6 (10.3.1)", "iPhone X"]                                                      |
| project                | ./FastlaneBugApp.xcodeproj                                                             |
| scheme                 | FastlaneBugApp                                                                         |
| derived_data_path      | /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxf  |
|                        | ijcexhcerfzxlwdc                                                                       |
| clean                  | false                                                                                  |
| skip_build             | false                                                                                  |
| output_directory       | ./fastlane/test_output                                                                 |
| output_types           | html,junit                                                                             |
| buildlog_path          | ~/Library/Logs/scan                                                                    |
| include_simulator_logs | false                                                                                  |
| open_report            | false                                                                                  |
| skip_slack             | false                                                                                  |
| slack_only_on_failure  | false                                                                                  |
| use_clang_report_name  | false                                                                                  |
| fail_build             | true                                                                                   |
| xcode_path             | /Applications/Xcode.app                                                                |
+------------------------+----------------------------------------------------------------------------------------+

[10:26:30]: [36m$ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -scheme FastlaneBugApp -project ./FastlaneBugApp.xcodeproj -destination 'platform=iOS Simulator,id=4BD7096A-68AA-4E58-900D-80CA87F5B823' -destination 'platform=iOS Simulator,id=AE0DC84B-6BD5-44B0-9EEE-1A00C5AD344C' -derivedDataPath '/Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc' build test | tee '/Users/terje.tjervaag/Library/Logs/scan/FastlaneBugApp-FastlaneBugApp.log' | xcpretty  --report html --output '/Users/terje.tjervaag/Development/FastlaneBugApp/fastlane/test_output/report.html' --report junit --output '/Users/terje.tjervaag/Development/FastlaneBugApp/fastlane/test_output/report.junit' --report junit --output '/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/junit_report20180226-72150-182e22m'[0m
[10:26:30]: ▸ [35mLoading...[0m
[10:26:31]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:26:31]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:26:31]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:26:32]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:26:33]: [35m[33m▸[0m [39;1mBuild[0m Succeeded[0m
[10:26:33]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:26:33]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:26:33]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:26:34]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:26:34]: [35m[33m▸[0m [39;1mBuild[0m Succeeded[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.625 xcodebuild[72190:2306132]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:26:34]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-5C2B169E-C28C-40AB-B97B-E63681A90AEB/FastlaneBugAppUITests-FCDA2508-F599-402B-BD00-B50EDAF0DF96/Session-FastlaneBugAppUITests-2018-02-26_102634-4wj1tK.log[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.626 xcodebuild[72190:2306021] [MT] IDETestOperationsObserverDebug: (63F455B4-5951-4383-92B5-5864620930A4) Beginning test session FastlaneBugAppUITests-63F455B4-5951-4383-92B5-5864620930A4 at 2018-02-26 10:26:34.626 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fa1df2f4b70> {[0m
[10:26:34]: ▸ [35mSimDevice: iPhone 6 (4BD7096A-68AA-4E58-900D-80CA87F5B823, iOS 10.3, Booted)[0m
[10:26:34]: ▸ [35m} (10.3.1 (14E8301))[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.635 xcodebuild[72190:2306132]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:26:34]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-5C2B169E-C28C-40AB-B97B-E63681A90AEB/FastlaneBugAppTests-F4BC64C7-C1D3-404B-A18D-329C4606ADD4/Session-FastlaneBugAppTests-2018-02-26_102634-gRlb4k.log[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.635 xcodebuild[72190:2306021] [MT] IDETestOperationsObserverDebug: (765CED27-C3C5-4165-94D4-7E007CC45E12) Beginning test session FastlaneBugAppTests-765CED27-C3C5-4165-94D4-7E007CC45E12 at 2018-02-26 10:26:34.635 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fa1df2f4b70> {[0m
[10:26:34]: ▸ [35mSimDevice: iPhone 6 (4BD7096A-68AA-4E58-900D-80CA87F5B823, iOS 10.3, Booted)[0m
[10:26:34]: ▸ [35m} (10.3.1 (14E8301))[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.910 xcodebuild[72190:2306203]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:26:34]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-0F2F9F30-D589-4FBE-A543-B43015D221B8/FastlaneBugAppUITests-5032E673-FF65-4EC7-B468-1B1E4FE5391C/Session-FastlaneBugAppUITests-2018-02-26_102634-lAs9Yq.log[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.910 xcodebuild[72190:2306021] [MT] IDETestOperationsObserverDebug: (BD3194A4-E63D-413F-B0C1-0F847AE22752) Beginning test session FastlaneBugAppUITests-BD3194A4-E63D-413F-B0C1-0F847AE22752 at 2018-02-26 10:26:34.910 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fa1df506ae0> {[0m
[10:26:34]: ▸ [35mSimDevice: iPhone X (AE0DC84B-6BD5-44B0-9EEE-1A00C5AD344C, iOS 11.2, Booted)[0m
[10:26:34]: ▸ [35m} (11.2 (15C107))[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.917 xcodebuild[72190:2306129]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:[0m
[10:26:34]: ▸ [35m/var/folders/1g/23yvyqf55n70_83_269tnvj00000gp/T/com.apple.dt.XCTest/IDETestRunSession-0F2F9F30-D589-4FBE-A543-B43015D221B8/FastlaneBugAppTests-C3A75F6E-F8A8-45B8-91A7-F28A02ACE703/Session-FastlaneBugAppTests-2018-02-26_102634-ANkSWu.log[0m
[10:26:34]: ▸ [35m2018-02-26 10:26:34.917 xcodebuild[72190:2306021] [MT] IDETestOperationsObserverDebug: (B4EE6B71-789B-46C0-91B3-0B52271E5A3D) Beginning test session FastlaneBugAppTests-B4EE6B71-789B-46C0-91B3-0B52271E5A3D at 2018-02-26 10:26:34.917 with Xcode 9C40b on target <DVTiPhoneSimulator: 0x7fa1df506ae0> {[0m
[10:26:34]: ▸ [35mSimDevice: iPhone X (AE0DC84B-6BD5-44B0-9EEE-1A00C5AD344C, iOS 11.2, Booted)[0m
[10:26:34]: ▸ [35m} (11.2 (15C107))[0m
[10:26:36]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugApp [Debug][0m
[10:26:36]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:26:36]: [35m[33m▸[0m [39;1mProcessing[0m Info.plist[0m
[10:26:38]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app[0m
[10:26:39]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugAppUITests [Debug][0m
[10:26:39]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:26:39]: [35m[33m▸[0m [39;1mBuilding[0m FastlaneBugApp/FastlaneBugAppTests [Debug][0m
[10:26:39]: [35m[33m▸[0m [39;1mCheck Dependencies[0m
[10:26:39]: [35m[33m▸[0m [39;1mLinking[0m FastlaneBugAppTests[0m
[10:26:40]: [35mRunning Tests: [33m▸[0m [39;1mTouching[0m FastlaneBugAppTests.xctest[0m
[10:26:40]: [35m[33m▸[0m [39;1mSigning[0m /Users/terje.tjervaag/Library/Developer/Xcode/DerivedData/FastlaneBugApp-aoqpxlkzrmxfijcexhcerfzxlwdc/Build/Products/Debug-iphonesimulator/FastlaneBugApp.app/PlugIns/FastlaneBugAppTests.xctest[0m
[10:27:12]: [35m[33m▸[0m [39;1mTest[0m Succeeded[0m
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 0 |
| Number of failures | [32m0[0m |
+--------------------+---+

+------+------------------+-------------+
|           [32mfastlane summary[0m            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 2    | run_tests        | 63          |
+------+------------------+-------------+

[10:27:27]: [32mfastlane.tools finished successfully 🎉[0m

</pre>
</details>

### Environment

<details>
  <pre>
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.3                                     |
| Ruby                        | 2.5.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.gems/bin/fastlane                        |
| Host                        | Mac OS X 10.13.3 (17D102)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.0/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2n  7 Dec 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Do a buggy test run"
  lane :test do
    run_tests(
      devices: [
        "iPhone 5 (9.3)",
        "iPhone 6 (10.3.1)"
      ]
    )
  end

  desc "Do a working test run"
  lane :workingtest do
    run_tests(
      devices: [
        "iPhone 6 (10.3.1)",
        "iPhone X"
      ]
    )
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
# apple_id("[[APPLE_ID]]") # Your Apple email address


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.82.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.2.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.6        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.60.0       |
| faraday                   | 0.14.0       |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.1.2        |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| openssl                   | 2.1.0        |
| date                      | 1.0.0        |
| fileutils                 | 1.0.2        |
| ipaddr                    | 1.2.0        |
</details>

*generated on:* **2018-02-26**
</details>
  </pre>
</details>

**Comment by terje** [comment]
I tested this using the latest Fastlane 2.89.0 now and the test still fails as described. I have updated the referenced GitHub repo to use this latest Fastlane version.

**Comment by jhandguy** [comment]
The same problem occurs to me as well, would be really nice to get this fixed.

**Comment by patreu22** [comment]
+1

**Comment by terje** [comment]
Updated Fastlane to 2.98.0 in the linked repo. The bug is still present.

**Comment by jhandguy** [comment]
Updated to 2.100.1 ➡️ still happening 😢

**Comment by Dahlgren** [comment]
Make sure that `ONLY_ACTIVE_ARCH = NO` (Build Active Architecture Only) is set for your project's targets, including test targets.

If you are using CocoaPods add the following to your Podfile,

```
post_install do |installer_representation|
    installer_representation.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['ONLY_ACTIVE_ARCH'] = 'NO'
        end
    end
end
```

The iPhone 4s with iOS 9.3 requires i386 while the newer iOS targets are using x86_64. By using the settings above both i386 and x86_64 will be built for the test targets.
